ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -1,352 +1,352 @@
+                ModelState.AddModelError("submissionText", "Your answer cannot be empty");
+            }
+
+            SubmissionText = submissionText;
+            if (DateTime.UtcNow < Event.EventBegin)
+            {
+                return NotFound("The event hasn't started yet!");
+            }
+
+            await SetupContext(puzzleId);
+
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+            
+            // Don't allow submissions if the team is locked out.
+            if (PuzzleState.IsTeamLockedOut)
+            {
+                return Page();
+            }
+
+            // Don't allow submissions if team is in email only mode.
             if (PuzzleState.IsEmailOnlyMode)
             {
                 return Page();
-            }
-
-            // Don't allow submissions after the answer has been found.
-            if (PuzzleState.SolvedTime != null)
-            {
-                return Page();
-            }
-
-            // Don't accept posted submissions when a puzzle is causing lockout
-            if (PuzzlesCausingGlobalLockout.Count != 0 && !PuzzlesCausingGlobalLockout.Contains(Puzzle))
-            {
-                return Page();
-            }
-
-            // Soft enforcement of duplicates to give a friendly message in most cases
-            DuplicateSubmission = (from sub in Submissions
-                                   where sub.SubmissionText == ServerCore.DataModel.Response.FormatSubmission(submissionText)
-                                   select sub).Any();
-
-            if (DuplicateSubmission)
-            {
-                return Page();
-            }
-
-            // Create submission and add it to list
-            Submission submission = new Submission
-            {
-                TimeSubmitted = DateTime.UtcNow,
-                Puzzle = PuzzleState.Puzzle,
-                Team = PuzzleState.Team,
-                Submitter = LoggedInUser,
-                AllowFreeformSharing = AllowFreeformSharing
-            };
-
-            string submissionTextToCheck = ServerCore.DataModel.Response.FormatSubmission(submissionText);
+            }
+
+            // Don't allow submissions after the answer has been found.
+            if (PuzzleState.SolvedTime != null)
+            {
+                return Page();
+            }
+
+            // Don't accept posted submissions when a puzzle is causing lockout
+            if (PuzzlesCausingGlobalLockout.Count != 0 && !PuzzlesCausingGlobalLockout.Contains(Puzzle))
+            {
+                return Page();
+            }
+
+            // Soft enforcement of duplicates to give a friendly message in most cases
+            DuplicateSubmission = (from sub in Submissions
+                                   where sub.SubmissionText == ServerCore.DataModel.Response.FormatSubmission(submissionText)
+                                   select sub).FirstOrDefault(null).?ResponseText;

```

Comment:
Should be `?.`, not `.?`.

[Also, the `null` is not strictly required here, but it's harmless to provide it so no worries either way.]

Reply:
Fixed. I didn't replace `null` because `FirstOrDefault` actually returns `default(T)` if it's unspecified and there are cases where that's not `null`. So good habit to be explicit here.

Reply:
Also, line 100 needs to be
`select sub).FirstOrDefault(null)?.Response.ResponseText;`

When I went to my copy of this file in the master branch to test out the line above, it did not complain about line endings, so I think that's an issue with whatever editor you're using. Do you mind if I make my own version of this PR with proper line endings?

Reply:
Yes, go ahead and replicate the changes in your own repo.

The inconsistent line endings are on lines 80-82 and 118-124. These lines are separated by `\r\n` rather than just `\n`, I did the edits directly on github.com although emacs also sees the extra characters.

Reply:
In working to replicate these changes, I find that they do not work for the case where the submission has no response (i.e. it's not a partial, it's just wrong).

Unless there is some reason this issue is more important for PH22 than it's been for previous events, I think the safest thing to do is to wait on this. This has never cropped up as an issue before.

Reply:
Except for freeform responses, this would just be replacing null with Incorrect. I don't want to break freeform responses although we're not using them in PH22.

The motivation to do this is that beta teams complained that it was tedious to have to search the page for a duplicate response. I don't know how important it is but it seemed easy. But apparently it isn't.

I've actually written a prototype of a new data checker that can run either in ajax mode with a server or independently without a server. You can check it out here. https://vroo.site44.com/demo/checker.html After this hunt, I'd like to see if I could splice it into puzzleserver.

Reply:
The issue is that the Response object is null, which crashes when I simply use Response.ResponseText. This can surely be worked around with some minor effort, but since it is trickier than expected to get the logic right, I am asking whether this is somehow more important than it has been for previous events. Trying to make an informed decision between benefit and risk.

Reply:
It's probably not worth any risk you think is significant. Just because beta teams complain about something doesn't mean we should jump. I thought it was easy and low risk. I will defer to your judgment that it isn't.

Reply:
@tabascq , I ended up solving this in my other PR
https://github.com/PuzzleServer/mainpuzzleserver/pull/758